src/task.c

```diff
@@ -686,14 +686,18 @@ static void record_backtrace(void)
 static jl_value_t *array_ptr_void_type = NULL;
 DLLEXPORT jl_value_t *jl_backtrace_from_here(void)
 {
-    if (array_ptr_void_type == NULL)
-        array_ptr_void_type = jl_apply_type((jl_value_t*)jl_array_type,
-                                            jl_tuple2(jl_voidpointer_type,
-                                                      jl_box_long(1)));
-    jl_array_t *bt = jl_alloc_array_1d(array_ptr_void_type, MAX_BT_SIZE);
+    jl_tuple_t *tp = NULL;
+    jl_array_t *bt = NULL;
+    JL_GC_PUSH3(&tp, &bt, &array_ptr_void_type);
+    if (array_ptr_void_type == NULL) {
+        tp = jl_tuple2(jl_voidpointer_type, jl_box_long(1));
+        array_ptr_void_type = jl_apply_type((jl_value_t*)jl_array_type, tp);
+    }
+    bt = jl_alloc_array_1d(array_ptr_void_type, MAX_BT_SIZE);
     size_t n = rec_backtrace((ptrint_t*)jl_array_data(bt), MAX_BT_SIZE);
```

Comment:
For `bt`, I was concerned about `jl_array_del_end` (which can call `array_resize_buffer` via `array_try_unshare`), not `rec_backtrace`.